app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -123,48 +180,57 @@ private void setupFocusedFavorListen() {
             getViewLifecycleOwner(),
             favor -> {
               try {
-                if (favor != null) {
+                if (favor != null && favorViewModel.isShowObservedFavor()) {
+                  favorViewModel.setShowObservedFavor(false);
                   focusedFavor = favor;
                   boolean isRequested = // check if favor is requested
                       favor
                           .getRequesterId()
                           .equals(DependencyFactory.getCurrentFirebaseUser().getUid());
-
-                  Marker marker = drawFavorMarker(focusedFavor, isRequested);
+                  boolean isEdited = focusedFavor.getStatusId() == FavorStatus.EDIT.toInt();
+                  Marker marker = drawFavorMarker(focusedFavor, isRequested, isEdited);
+                  if (isEdited) newMarkers.add(marker);
                   marker.showInfoWindow();
-                  focusViewOnLocation(focusedFavor.getLocation());
+                  focusViewOnLocation(focusedFavor.getLocation(), true);
                 }
               } catch (Exception e) {
                 CommonTools.showSnackbar(requireView(), getString(R.string.error_database_sync));
               }
             });
   }
 
-  private Marker drawFavorMarker(Favor favor, boolean isRequested) {
+  private Marker drawFavorMarker(Favor favor, boolean isRequested, boolean isEdited) {
     LatLng latLng =
         new LatLng(favor.getLocation().getLatitude(), favor.getLocation().getLongitude());
     float markerColor =
         isRequested ? BitmapDescriptorFactory.HUE_AZURE : BitmapDescriptorFactory.HUE_RED;
+    String markerTitle =
+        (isEdited && favor.getTitle().equals(""))
+            ? "Drag marker to choose position"
+            : favor.getTitle();
+    String markerDescription = isEdited ? "Click window to request favor" : favor.getDescription();
```

Comment:
Should maybe make it explicit that the marker is draggable. Should we replace the text with "Drag marker to desired location"?

Reply:
done